WPMedia/fusion-news-theme-blocks
# Tmedia 491 offer feature

## Description
Adds the Offer feature to be used on an offer page

## Jira Ticket
https://arcpublishing.atlassian.net/browse/TMEDIA-491

## Acceptance Criteria
Just see the ticket please.

## Test Steps
- Add test steps a reviewer must complete to test this PR

1. Checkout this branch `git checkout tmedia-491-offer-feature`
2. Run fusion repo with linked blocks `npx fusion start -f -l @wpmedia/subscriptions-block,@wpmedia/identity-block,@wpmedia/header-nav-chain-block,@wpmedia/shared-styles`
3. In PB, create a new page called `offer`
4. Choose right rail layout.
5. In the nav section, add Navigation block and under that the Identity block
6. In the fullwidth section add the Offer block.
7. For the offer block prop types, set campaignCode to allaccess, loginURL to /login/ and checkoutURL to /checkout/
8. Publish page.
9. Go to published page for the Gazzette.
10. You should see 4 offers.
11. Now change the url in the browser to http://localhost/pf/offer/?_website=the-gazette&campaign=sports and now you should only see two offers


## Dependencies or Side Effects
Please note the unfortunate change to the buttons.  Had to add a param to tell if the text contains HTML and if so to render dangerously.  Content from Subs usually has HTML content.

## Author Checklist
_The author of the PR should fill out the following sections to ensure this PR is ready for review._
- [x ] Confirmed all the test steps a reviewer will follow above are working.
- [ x] Confirmed there are no linter errors. Please run `npm run lint` to check for errors. Often, `npm run lint:fix` will fix those errors and warnings.
- [ x] Ran this code locally and checked that there are not any unintended side effects. For example, that a CSS selector is scoped only to a particular block.
- [ x] Confirmed this PR has reasonable code coverage. You can run `npm run test:coverage` to see your progress.
  - [x ] Confirmed this PR has unit test files
  - [x ] Ran `npm run test`, made sure all tests are passing


## Reviewer Checklist
_The reviewer of the PR should copy-paste this template into the review comments on review._
